tests/PuliServiceProviderTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the puli/silex-provider package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\SilexProvider\Tests;
+
+use PHPUnit_Framework_TestCase;
+use Puli\SilexProvider\PuliServiceProvider;
+use Silex\Application;
+use Silex\Provider\TwigServiceProvider;
+
+class PuliServiceProviderTest extends PHPUnit_Framework_TestCase
+{
+    public function testConfiguredApplication()
+    {
+        $app = new Application();
+        $app->register(new PuliServiceProvider());
+
+        require_once __DIR__.'/Fixtures/.puli/GeneratedPuliFactory.php';
+        $app['puli.factory'] = new \Puli\GeneratedPuliFactory();
```

Comment:
Instead of using a real factory class, I'd use a mock here and then test that the right method is called. I'd create one test per service.